lib/beaker/hypervisor/openstack.rb

```diff
@@ -91,6 +91,15 @@ def provision
         if @options[:openstack_keyname]
           @logger.debug "Adding optional key_name #{@options[:openstack_keyname]} to #{host.name} (#{host[:vmhostname]})"
           options[:key_name] = @options[:openstack_keyname]
+        else
+          @logger.debug "Generate a new rsa key"
```

Comment:
The openstack.provision method is getting pretty lengthy.  Let's get setting the keys as a method that can be called during provisioning to better encapsulate the code.